src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -207,6 +188,68 @@ public String getScript() {
         return c;
     }
 
+    private String binaryLauncherCmd(ShellController c, FilePath ws, OsType os, String interpreter, String scriptPath, String cookieValue, String cookieVariable) throws IOException, InterruptedException {
+        String logFile = c.getLogFile(ws).toURI().getRawPath();
+        String resultFile = c.getResultFile(ws).toURI().getRawPath();
+        FilePath controlDir = c.controlDir(ws);
+        String outputFile = c.getOutputFile(ws).toURI().getRawPath();
+
+        String launcher_binary = LAUNCHER_PREFIX + os.toString();
+
+        InputStream launcherStream = this.getClass().getResourceAsStream(launcher_binary);
```

Comment:
As a matter of style (consider a subclass in another package):

```suggestion
        InputStream launcherStream = BourneShellScript.class.getResourceAsStream(launcher_binary);
```